### ISSUE TITLE
Remove development related CLI commands

### ISSUE THREAD

**Issue by merelcht** [kedro-org/kedro#1617]
Following the research conducted in https://github.com/kedro-org/kedro/issues/1293 it has been decided that the following commands will all be deprecated and removed:

1. `kedro jupyter convert`
2. `kedro activate-nbstripout`
3. `kedro build-docs`
4. `kedro build-reqs`
5. `kedro lint`
6. `kedro test`

Remove all the above commands from Kedro on `develop`, this also includes updating documentation that mentions any of the above commands. 

The template project should still be linted and `pytest` should run on it, so some of these calls (e.g. in e2e tests) need to be replaced with the underlying calls, e.g. `pytest`.

This will be released in Kedro `0.19.0`

**Comment by stichbury** [comment]
Just wanted to put a note here so it doesn't get lost. I've updated the docs in advance of the removal of `build-docs` in #2304 but once we reach the point of removing `build-docs` we need to revisit those docs and also do the following:

* Remove the code for build-docs (obvious)
* Remove the docs dependencies in setup.py:
```
"docs": [
            "docutils<0.18.0",
            "sphinx~=3.4.3",
            "sphinx_rtd_theme==0.5.1",
            "nbsphinx==0.8.1",
            "nbstripout~=0.4",
            "sphinx-autodoc-typehints==1.11.1",
            "sphinx_copybutton==0.3.1",
            "ipykernel>=5.3, <7.0",
            "Jinja2<3.1.0",
            "myst-parser~=0.17.2",
        ]
```

* Update the project template/starter to remove the `docs/source` subfolder so there's no existing `conf.py` and `index.rst` to confuse things (then we can update these docs again).
* Potentially, add basic documentation into the project template as to how to create docs (put a markdown folder in the /docs folder maybe) that echoes this advice on using Sphinx.

**Comment by noklam** [comment]
Closed by #3065
